File: app/src/main/java/ru/gsench/githubusers/presentation/view/CoordinatorView.java
package ru.gsench.githubusers.presentation.view;

import java.net.URL;

import ru.gsench.githubusers.presentation.presenter.UserListPresenter;
import ru.gsench.githubusers.presentation.presenter.UserPresenter;

/**
 * Created by grish on 09.05.2017.
 */

public interface CoordinatorView {
    void init();
    void openSearchView();
    void openUserList(UserListPresenter presenter);
    void openUser(UserPresenter presenter);
    void openBrowser(URL url);
}

File: app/src/main/java/ru/gsench/githubusers/presentation/view/CoordinatorView.kt
package ru.gsench.githubusers.presentation.view

import java.net.URL

import ru.gsench.githubusers.presentation.presenter.UserListPresenter
import ru.gsench.githubusers.presentation.presenter.UserPresenter

/**
 * Created by grish on 09.05.2017.
 */

interface CoordinatorView {
    fun init()
    fun openSearchView()
    fun openUserList(presenter: UserListPresenter)
    fun openUser(presenter: UserPresenter)
    fun openBrowser(url: URL)
    fun closeView()
    fun closeUser()
    val isUserOpened: Boolean
}
